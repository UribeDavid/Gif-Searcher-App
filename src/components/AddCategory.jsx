import { useState } from 'react'

import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = ({target}) => setInputValue(target.value)

    const handleSubmit = (evt) => {
        evt.preventDefault();
        if (!inputValue.length) return;
        onNewCategory( inputValue.trim() );
        setInputValue('');
    }

    return (
        <form onSubmit={ handleSubmit } aria-label="form">
            <input
                type="text"
                placeholder="Buscar gifs..."
                value={inputValue}
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}