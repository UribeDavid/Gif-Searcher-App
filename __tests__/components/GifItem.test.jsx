import { render, screen } from "@testing-library/react"

import { GifItem } from "../../src/components/GifItem"

describe('Tests in <GifItem/>', () => {

    const [title, url] = ['Test title', 'https://www.test.com/']

    test('Should match with the snapshot', () => {
        const { container } = render(
            <GifItem title={ title } url={ url }/>
        )
        expect( container ).toMatchSnapshot();
    })

    test('Should show the image with the URL and the ALT indicated', () => {
        render(
            <GifItem title={ title } url={ url }/>
        )
        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe(url)
        expect( alt ).toBe(title)
    })

    test('Should show title in the component', () => {
        render(
            <GifItem title={ title } url={ url }/>
        )
        expect( screen.getByText(title) ).toBeTruthy();
    })
})