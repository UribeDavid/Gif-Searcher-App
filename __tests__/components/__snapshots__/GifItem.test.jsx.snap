// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifItem/> Should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test title"
      src="https://www.test.com/"
    />
    <p>
      Test title
    </p>
  </div>
</div>
`;
